great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1690,7 +1691,10 @@ def delete_datasource(
 
         datasource = self.get_datasource(datasource_name=datasource_name)
 
-        if save_changes and not isinstance(datasource, FluentDatasource):
+        if isinstance(datasource, FluentDatasource):
+            # Note: this results in some unnecessary dict lookups
+            self._delete_fluent_datasource(datasource_name)
```

Comment:
Seems like we should also call `context._save_project_config()` at the end of this method otherwise the changes don't get reflected in the `great_expectations.yml`.